cli/src/cli_util.rs

```diff
@@ -2292,34 +2292,45 @@ pub struct EarlyArgs {
     pub config_toml: Vec<String>,
 }
 
-/// `-m/--message` argument which should be terminated with `\n`.
+/// Create a description from a list of paragraphs.
 ///
 /// Based on the Git CLI behavior. See `opt_parse_m()` and `cleanup_mode` in
 /// `git/builtin/commit.c`.
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct DescriptionArg(String);
-
-impl DescriptionArg {
-    pub fn as_str(&self) -> &str {
-        self.0.as_str()
-    }
-}
-
-impl From<String> for DescriptionArg {
-    fn from(s: String) -> Self {
-        DescriptionArg(text_util::complete_newline(s))
-    }
+pub fn join_message_paragraphs(paragraphs: &[String]) -> String {
+    text_util::complete_newline(paragraphs.join("\n\n"))
 }
 
-impl From<&DescriptionArg> for String {
-    fn from(arg: &DescriptionArg) -> Self {
-        arg.0.to_owned()
-    }
+/// `-m/--message` arguments
+///
+/// Based on the Git CLI behavior. See `opt_parse_m()` and `cleanup_mode` in
+/// `git/builtin/commit.c`.
+#[derive(clap::Args, Debug, Clone)]
+pub struct DescriptionArgs {
+    /// The change description to use (don't open editor)
```

Comment:
Note that this also changes commands which _don't_ open the editor by default to include this in the `--help` output. But it seems worth it to me, to allow message cleanup to be specified in one place, for example.

But it would be easy enough to distribute a Vec<String> to each command, and then we can document each differently.

Reply:
Since we already have a doc comment where we use the `DescriptionArgs`, does this even get rendered anywhere? Oh, you did remove it from `CommitArgs`. If possible, I'd rather remove it from here and add it back on `CommitArgs`. Will that result in each command having its own `--help` text?

Reply:
Ah, it's the other way around, the doc comment on `DescriptionArgs` is unused for the CLI (after all, `DescriptionArgs could contain multiple flags, and how would it know which flag the doc comment was for?

I'm feeling more like it makes sense to not use an `Args` derive, and just provide a function to take a list of message paragraphs, and keep defining the flag for each command that uses it.

Reply:
> Ah, it's the other way around, the doc comment on `DescriptionArgs` is unused for the CLI (after all, `DescriptionArgs could contain multiple flags, and how would it know which flag the doc comment was for?

Isn't that what we want, so we can get different messages for each command?

> I'm feeling more like it makes sense to not use an `Args` derive, and just provide a function to take a list of message paragraphs, and keep defining the flag for each command that uses it.

That's fine with me either way.

Reply:
No, sorry, I realize that's still not super clear. The doc comments on the fields of type `DescrptionArgs` are not used. The doc comments on the fields of `DescrptionArgs` itself are the ones that matter.

Reply:
Ah, that makes sense. Now I see what you meant with "how would it know which flag the doc comment was for?" :)

The free function you were thinking about still sounds good.